### ISSUE TITLE
Multicast source address affected by `weave expose`

### ISSUE THREAD

**Issue by bboreham** [weaveworks/weave#2147]
I have a small demo program that prints the hostname and source address of other multicasters.

````
host1$ weave launch 192.168.48.12
host2$ weave launch 192.168.48.11
host1$ weave run --name=c1 bboreham/mweb
host2$ weave run --name=c2 bboreham/mweb
host1$ curl `docker inspect -f {{.NetworkSettings.IPAddress}} c1`:8080
c1.weave.local 10.32.0.1:57206
c2.weave.local 10.40.0.0:45872
host2$ curl `docker inspect -f {{.NetworkSettings.IPAddress}} c2`:8080
c1.weave.local 10.32.0.1:57206
c2.weave.local 10.40.0.0:45872
````

So far so good.  Run `weave expose`, and nothing changes:

````
host1$ weave expose
10.32.0.2
host1$ curl `docker inspect -f {{.NetworkSettings.IPAddress}} c1`:8080
c1.weave.local 10.32.0.1:57206
c2.weave.local 10.40.0.0:45872
````

But now run a new container:

````
host1$ weave run --name=c3 bboreham/mweb
266568a679e4252f75a9617ea67ecb2fab610f5e6ae139018ec1edf0e273cf2f
host1$ curl `docker inspect -f {{.NetworkSettings.IPAddress}} c1`:8080
c1.weave.local 10.32.0.1:57206
c2.weave.local 10.40.0.0:45872
c3.weave.local 10.32.0.2:53622
````

so now `c3` is being reported at the `weave expose` address.  Same on the other host:

````
host2$ curl `docker inspect -f {{.NetworkSettings.IPAddress}} c2`:8080
c2.weave.local 10.40.0.0:45872
c3.weave.local 10.32.0.2:53622
c1.weave.local 10.32.0.1:57206
````

`weave hide` puts it back to normal again:

````
host1$ weave hide
10.32.0.2
host1$ curl `docker inspect -f {{.NetworkSettings.IPAddress}} c1`:8080
c1.weave.local 10.32.0.1:57206
c2.weave.local 10.40.0.0:45872
c3.weave.local 10.32.0.3:53622
````

**Comment by bboreham** [comment]
Compare #605

**Comment by bboreham** [comment]
Fixed by #2158
